src/libs/DistanceRequestUtils.ts

```diff
@@ -116,15 +116,19 @@ function getRoundedDistanceInUnits(distanceInMeters: number, unit: Unit): string
  */
 function getRateForDisplay(
     hasRoute: boolean,
-    unit: Unit,
+    unit: Unit | undefined,
     rate: number | undefined,
     currency: string | undefined,
     translate: LocaleContextProps['translate'],
     toLocaleDigit: LocaleContextProps['toLocaleDigit'],
+    isOffline?: boolean,
 ): string {
-    if (!hasRoute || !rate || !currency) {
+    if (isOffline) {
```

Comment:
This check is too simple. We should only show this is we're offline and don't have the rates loaded for the workspace. We can still show the proper rate while offline if the workspace rates are loaded, or if it's a P2P distance request.